nixos/modules/programs/singularity.nix

```
@@ -3,32 +3,83 @@
 with lib;
 let
   cfg = config.programs.singularity;
-  singularity = pkgs.singularity.overrideAttrs (attrs : {
-    installPhase = attrs.installPhase + ''
-      mv $out/libexec/singularity/bin/starter-suid $out/libexec/singularity/bin/starter-suid.orig
-      ln -s /run/wrappers/bin/singularity-suid $out/libexec/singularity/bin/starter-suid
-    '';
-  });
-in {
+in
+{
   options.programs.singularity = {
-    enable = mkEnableOption "Singularity";
+    enable = mkEnableOption "singularity" // {
+      description = ''
+        Whether to install Singularity/Apptainer
+        and add SUID support.
+      '';
+    };
+    package = mkOption {
+      default = pkgs.singularity;
+      defaultText = "pkgs.singularity";
+      description = ''
+        Singularity package to override and install. Typical values are
+        pkgs.singularity, pkgs.singularity-legacy, pkgs.apptainer, and pkgs.singularity-ce.
+      '';
+      type = types.package;
+    };
+    packageOverriden = mkOption {
+      default = mkDefault null;
```

```suggestion
      default = null;
```
default is already the default and can be overwriten in config.